### PR TITLE
[RELEASE] featurebyte 1.1.0 breaking change on feature job settings

### PR DESCRIPTION
**Pull request by bushwhackr**
## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -109,7 +109,7 @@ databricks-sdk = { version = "^0.24.0", optional = true }
 databricks-sql-connector = { version = "^3.1.1", optional = true }
 fastapi = { version =  "^0.96.0", optional = true }  # Need to bump pydantic to 2.X.X
 feast = { version = "^0.35.0", optional = true, extras = ["snowflake", "redis", "mysql"]}
-featurebyte-freeware = { version = "^0.2.16", optional = true }
+featurebyte-freeware = { version = "^0.2.18", optional = true }
```
Comment (gee-senbong) [inline pyproject.toml:112]: @kchua78 @bushwhackr From the error, seems this commit (https://github.com/featurebyte/freeware/commit/fc96716582edaccfe4fe4fc3736d0fae32e35443) is not included. Should we bump the freeware to higher version?

File: pyproject.toml [hunk 2]
```diff
@@ -134,7 +134,7 @@ requests-kerberos = { version = "^0.14.0", optional = true }
 rich = "^13.3.4"
 sasl = { version = "^0.3.1", optional = true }
 smart-open = { version = "^6.3.0", extras = ["azure", "gcs"], optional = true }
-snowflake-connector-python = { version = "^3.7.1", optional = true }
+snowflake-connector-python = { version = "3.7.1", optional = true }  # Breaking ResultMetadata does not have fields
```
Comment (gee-senbong) [inline pyproject.toml:137]: @kchua78 I suspect the fixtures you added using the freeware including this commit (https://github.com/featurebyte/freeware/pull/95/files) but current `0.2.18` freeware version does not include that commit.

**Review by gee-senbong — APPROVED** [review]

File: featurebyte/worker/task/feature_job_setting_analysis_backtest.py [hunk 3]
```diff
@@ -94,9 +94,9 @@ async def execute(self, payload: FeatureJobSettingAnalysisBackTestTaskPayload) -
         # run backtest
         await self.task_progress_updater.update_progress(percent=5, message="Running Analysis")
         feature_job_setting = FeatureJobSetting(
-            frequency=payload.period,
+            period=payload.period,
```
Comment (kchua78) [inline featurebyte/worker/task/feature_job_setting_analysis_backtest.py:97]: @gee-senbong I think we don't need to modify `recommended_feature_job_setting` anymore

Comment (gee-senbong) [reply featurebyte/worker/task/feature_job_setting_analysis_backtest.py:97]: `FeatureJobSetting` expects `period` and `offset` parameters in https://github.com/featurebyte/freeware/blob/main/featurebyte_freeware/feature_job_analysis/schema.py#L352-L353.

File: tests/unit/worker/task/test_feature_job_setting_analysis.py [hunk 4]
```diff
@@ -71,6 +72,19 @@ def use_mock_event_dataset(self, mock_event_dataset):
...
+        """
+        Strip img tag from html
+        """
+        soup = bs4.BeautifulSoup(html, "html.parser")
+        for tag in soup.find_all("img"):
+            # strip image tag with base64 encoded image to avoid issue where different OS may
+            # generate different base64 encoded image (could be due to different font rendering)
+            if tag["src"].startswith("data:image/png;base64"):
+                tag.decompose()
+        return str(soup)
```
Comment (gee-senbong) [inline tests/unit/worker/task/test_feature_job_setting_analysis.py:86]: @kchua78 FYI, I update the test here to exclude the image tags from the html as the generated images are slightly different between Linux & MacOS.

File: featurebyte/worker/task/feature_job_setting_analysis_backtest.py (hunk 3 above)
Comment (kchua78) [reply featurebyte/worker/task/feature_job_setting_analysis_backtest.py:97]: I am referring to the `from_dict` line above this change. Support for the old parameters `frequency` and `job_time_modulo_frequency` has been added here:
https://github.com/featurebyte/freeware/blob/main/featurebyte_freeware/feature_job_analysis/schema.py#L377

**Pull request merged**